From e7f6388c110a8d352bbd4fe475624102a9a8ee47 Mon Sep 17 00:00:00 2001
From: Joseph Guo <qijian.guo@nxp.com>
Date: Thu, 17 Apr 2025 15:01:29 +0900
Subject: [PATCH 03/56] LF-15222 arm64: dts: imx8mp-frdm: add os08a20 support

add imx8mp-frdm-os08a20.dts and imx8mp-frdm-os08a20-dual.dts

Signed-off-by: Joseph Guo <qijian.guo@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |   1 +
 .../freescale/imx8mp-frdm-os08a20-dual.dts    | 110 ++++++++++++++++++
 .../dts/freescale/imx8mp-frdm-os08a20.dts     |  58 +++++++++
 3 files changed, 169 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20-dual.dts
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index db8130ed0559..5717d4964632 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -257,6 +257,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk-revb4-rpmsg.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk-revb4-rpmsg-lpv.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm-os08a20.dtb imx8mp-frdm-os08a20-dual.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-evk.dtb imx8mq-evk-rpmsg.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mq-evk.dtb imx8mq-evk-rpmsg.dtb imx8mq-evk-pcie1-m2.dtb imx8mq-evk-usd-wifi.dtb \
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20-dual.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20-dual.dts
new file mode 100644
index 000000000000..1bbb4d1f6ade
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20-dual.dts
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx8mp-frdm.dts"
+
+&i2c2 {
+	/delete-node/ap1302_mipi_0@3c;
+
+	os08a20_0: os08a20_mipi@36 {
+		compatible = "ovti,os08a20_mipi";
+		reg = <0x36>;
+		pinctrl-names = "default";
+		clocks = <&clk_dummy>;
+		clock-names = "csi_mclk";
+		csi_id = <0>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		status = "okay";
+
+		port {
+			os08a20_mipi_0_ep: endpoint {
+				data-lanes = <1 2 3 4>;
+				clock-lanes = <0>;
+				max-pixel-frequency = /bits/ 64 <266000000>;
+				remote-endpoint = <&mipi_csi0_ep>;
+			};
+		};
+
+	};
+};
+
+&i2c3 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>;
+	status = "okay";
+
+	os08a20_1: os08a20_mipi@36 {
+		compatible = "ovti,os08a20_mipi";
+		reg = <0x36>;
+		clocks = <&clk_dummy>;
+		clock-names = "csi_mclk";
+		csi_id = <1>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		status = "okay";
+
+		port {
+			os08a20_mipi_1_ep: endpoint {
+				data-lanes = <1 2 3 4>;
+				clock-lanes = <0>;
+				max-pixel-frequency = /bits/ 64 <266000000>;
+				remote-endpoint = <&mipi_csi1_ep>;
+			};
+		};
+	};
+};
+
+&cameradev {
+	status = "okay";
+};
+
+&isi_0 {
+	cap_device {
+		status = "disabled";
+	};
+};
+
+&isp_0 {
+	status = "okay";
+};
+
+&isp_1 {
+	status = "okay";
+};
+
+&dewarp {
+	status = "okay";
+};
+
+&mipi_csi_0 {
+	status = "okay";
+	clock-frequency = <266000000>;
+	assigned-clock-parents = <&clk IMX8MP_SYS_PLL1_266M>;
+	assigned-clock-rates = <266000000>;
+	/delete-node/port;
+	port {
+		mipi_csi0_ep:endpoint {
+			remote-endpoint = <&os08a20_mipi_0_ep>;
+			data-lanes = <4>;
+			csis-hs-settle = <16>;
+		};
+	};
+};
+
+&mipi_csi_1 {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	port@1 {
+		reg = <1>;
+		mipi_csi1_ep: endpoint {
+			remote-endpoint = <&os08a20_mipi_1_ep>;
+			data-lanes = <4>;
+			csis-hs-settle = <16>;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20.dts
new file mode 100644
index 000000000000..0ec2b8da13e9
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20.dts
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx8mp-frdm.dts"
+
+&i2c2 {
+	/delete-node/ap1302_mipi_0@3c;
+
+	os08a20_0: os08a20_mipi@36 {
+		compatible = "ovti,os08a20_mipi";
+		reg = <0x36>;
+		pinctrl-names = "default";
+		clocks = <&clk_dummy>;
+		clock-names = "csi_mclk";
+		csi_id = <0>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		status = "okay";
+
+		port {
+			os08a20_mipi_0_ep: endpoint {
+				data-lanes = <1 2 3 4>;
+				clock-lanes = <0>;
+				max-pixel-frequency = /bits/ 64 <500000000>;
+				remote-endpoint = <&mipi_csi0_ep>;
+			};
+		};
+
+	};
+};
+
+&isi_0 {
+	cap_device {
+		status = "disabled";
+	};
+};
+
+&isp_0 {
+	status = "okay";
+};
+
+&dewarp {
+	status = "okay";
+};
+
+&mipi_csi_0 {
+	status = "okay";
+
+	port {
+		mipi_csi0_ep: endpoint {
+			remote-endpoint = <&os08a20_mipi_0_ep>;
+			data-lanes = <4>;
+			csis-hs-settle = <16>;
+		};
+	};
+};
-- 
2.34.1

